src/app/features/rpc-transaction-request/use-rpc-transaction-request.tsx

```diff
@@ -9,23 +9,47 @@ import { analytics } from '@shared/utils/analytics';
 import { focusTabAndWindow } from '@app/common/focus-tab';
 import { useRpcRequestParams } from '@app/common/hooks/use-rpc-request-params';
 
-export interface RpcTransactionRequest {
-  origin: string;
-  requestId: string;
-  tabId: number;
+interface TransactionState {
   isBroadcasting: boolean;
   isLoading: boolean;
   isSubmitted: boolean;
+}
+
+type TransactionAction =
+  | { type: 'SET_BROADCASTING'; payload: boolean }
+  | { type: 'SET_LOADING'; payload: boolean }
+  | { type: 'SET_SUBMITTED'; payload: boolean };
+
+const initialState: TransactionState = {
+  isBroadcasting: false,
+  isLoading: false,
+  isSubmitted: false,
+};
+
+function transactionReducer(state: TransactionState, action: TransactionAction): TransactionState {
+  switch (action.type) {
+    case 'SET_BROADCASTING':
+      return { ...state, isBroadcasting: action.payload };
+    case 'SET_LOADING':
+      return { ...state, isLoading: action.payload };
+    case 'SET_SUBMITTED':
+      return { ...state, isSubmitted: action.payload };
+    default:
+      return state;
+  }
+}
```

Comment:
[This violates this best practice](https://redux.js.org/style-guide/#model-actions-as-events-not-setters). 

Can't this be modelled as a flow? Can all these values be true at once? If not, how can we write this reducer to never allow that state? Better that the dispatched action sets the state, rather than having it passed in as a payload.

Reply:
Yep, sure, I can do that way.

Reply:
`SET...` is used often in mobile btw. https://github.com/leather-io/mono/blob/dev/apps/mobile/src/features/send/send-flow-provider.tsx

Reply:
Looking at this a bit closer, maybe switching to the reducer is a bit overkill until the other transaction status flows are implemented. It feels a bit unnecessary to be tracking a user flow rn? I'd like to reevaluate the action button implementation, but I want to get the `stx_transferSip9Nft` refactored to Approver first.

Reply:
Even if more verbose, I think this is a better way of doing it with a reducer, though indeed if all states can be represented as a single string literal, a single usestate is fine. 

My initial comment really stems from seeing various useStates next to each other which is usually a bad sign. 

cc/ @tigranpetrossian on the state pattern used in mobile. Just renaming the events to be less "setter-y" avoids setter thinking. When is the account set... When the user chooses an account... thus the event is `ACCOUNT_SELECTED`

Reply:
Agree, thanks for suggesting the reducer. I lost track of the best practice not having used a `useReducer` in a while. It was good to think thru it, but I think I'll come back to it if needed later and just use one `useState` for now. I'm not sure the loading state even needs to be included here ...will finalize it today.